src/ast/rewriter/arith_rewriter.cpp

```diff
@@ -800,107 +800,9 @@ br_status arith_rewriter::mk_idiv_core(expr * arg1, expr * arg2, expr_ref & resu
         result = m_util.mk_numeral(div(v1, v2), is_int);
         return BR_DONE;
     }
-    if (m_util.is_numeral(arg2, v2, is_int) && v2.is_one()) {
-        result = arg1;
-        return BR_DONE;
-    }
-    if (m_util.is_numeral(arg2, v2, is_int) && v2.is_zero()) {
-        return BR_FAILED;
-    }
-    if (arg1 == arg2) {
-        expr_ref zero(m_util.mk_int(0), m());
-        result = m().mk_ite(m().mk_eq(arg1, zero), m_util.mk_idiv(zero, zero), m_util.mk_int(1));
-        return BR_REWRITE3;
-    }
-    if (divides(arg1, arg2, result)) {
-        return BR_REWRITE_FULL;
-    }
     return BR_FAILED;
 }
 
-
-// 
-// implement div ab ac = floor( ab / ac) = floor (b / c) = div b c
-//
-bool arith_rewriter::divides(expr* num, expr* den, expr_ref& result) {
```

Comment:
ditto, add this to rebase

Reply:
If I add this code it would not compile.